auth/auth/auth_utils.py

```diff
@@ -0,0 +1,16 @@
+import re
+from typing import Optional
+
+from .exceptions import AuthUserError
+
+
+def validate_credentials_secret_name_input(secret_name: Optional[str]):
+    if secret_name is None:
+        return
+
+    regex = re.compile(r'^[a-z0-9]([.\-]?[a-z0-9])*[a-z0-9]?$')
+    if not regex.match(secret_name):
+        raise AuthUserError(
+            f'invalid credentials_secret_name {secret_name}. Must match RFC1123 (lowercase alphanumeric plus "." and "-", start and end with alphanumeric)',
```

Comment:
Do we want to include the secret name in the error? Is that OK?

Reply:
Hmm, a couple of mitigating factors - 

1. This is just a proposed name, not an actual existing secret
2. There's no lookup into the system happening here. We're just pre-validating a string value that was supplied by the user

So I would argue that this is fine, unless you're unconvinced